wcfsetup/install/files/lib/data/user/UserProfile.class.php

```diff
@@ -352,7 +352,11 @@ public function getAvatar() {
 	 * @return	boolean
 	 */
 	public function canSeeAvatar() {
-		return (WCF::getUser()->userID == $this->userID || WCF::getSession()->getPermission('user.profile.avatar.canSeeAvatars'));
+		return (
+			WCF::getUser()->userID == $this->userID ||
+			WCF::getSession()->getPermission('user.profile.avatar.canSeeAvatars') ||
+			(($pending = WCF::getSession()->getPendingUserChange()) && $pending->userID == $this->userID)
```

Comment:
Why do I need to see the Avatars then? I assume that this is a change that is now irrelevant due to template changes...?

Reply:
Yes, this is technically irrelevant now. However it allows grants access to the pending user's avatar which is not wrong, so I'll leave this as-is.